src/core/webview/ClineProvider.ts

```diff
@@ -1740,7 +1749,17 @@ export class ClineProvider
 	 * https://www.eliostruyf.com/devhack-code-extension-storage-options/
 	 */
 
-	async getState() {
+	async getState(): Promise<
+		Omit<
+			ExtensionState,
+			| "clineMessages"
+			| "renderContext"
+			| "hasOpenedModeSelector"
+			| "version"
+			| "shouldShowAnnouncement"
+			| "hasSystemPromptOverride"
```

Comment:
I think it's a bit strange that `getState` omits certain fields and relies on `getStateToPostToWebview` to fill in what's missing, but this will get us on a better path to strict type safety and we can clean this up later.